.github/workflows/main.yml

```diff
@@ -0,0 +1,127 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - '*_maintenance'
+  pull_request:
+    branches:
+      - '*'
+  release:
+      types: [published]
+
+jobs:
+
+  build:
+
+    strategy:
+
+      # Don't cancel other jobs in the build matrix if one job fails.
+      fail-fast: false
+
+      matrix:
+
+        # Rather than generate all permutations of various settings,
+        # we want to explicitly list each of the variants we want to
+        # test. We can use `name` to declare the names of our variants,
+        # and then use `include` to define their settings.
+
+        name: [
+          linux-gcc11,
+          macos-arm64,
+        ]
+
+        include:
+
+          - name: linux-gcc11
+            os: ubuntu-20.04
+            publish: true
+            containerImage: ghcr.io/gafferhq/build/build:3.0.0
+
+          - name: macos-arm64
+            os: macos-14
+            publish: true
+            containerImage:
+            jobs: 3
+
+    runs-on: ${{ matrix.os }}
+
+    container: ${{ matrix.containerImage }}
+
+    env:
+      DEPENDENCIES_BUILD_DIR: "./build"
+    steps:
+
+    - uses: actions/checkout@v4
+
+    - name: Install toolchain (Linux)
+      run: |
+        # The Docker container configures bash shells such that they enable the
+        # software collections we want. If we could set GitHub's
+        # `defaults.run.shell` to `bash` then all our build steps would pick up
+        # this environment automatically. But we can't do that because it will
+        # break future Windows builds, and we can't configure a different shell
+        # per platform because GitHub won't allow it. But we can run _this_
+        # Linux-only step in bash, and transfer the environment out to be used
+        # in later steps.
+        echo $PATH > $GITHUB_PATH
+        echo LD_LIBRARY_PATH=$LD_LIBRARY_PATH >> $GITHUB_ENV
+      shell: bash
+      if: runner.os == 'Linux'
+
+    - name: Install toolchain (macOS)
+      run: |
+        # Choose the earliest Xcode version available on a macos-14 runner.
+        sudo xcode-select -s /Applications/Xcode_14.3.1.app/Contents/Developer
+        # Make homebrew's Python 3.11 available on `PATH`.
```

Comment:
Is there a problem with the system-level Python 3? I'm a bit wary of Homebrew - it was often the source of sudden build breaks in the past, when packages were removed with little notice.